completions/system.completion.sh

```diff
@@ -1,27 +1,44 @@
 #! bash oh-my-bash.module
 
-# Loads the system's Bash completion modules.
-# If Homebrew is installed (OS X), its Bash completion modules are loaded.
+# Loads one of the system's Bash-completion modules.
 
-if [ -f /etc/bash_completion ]; then
-  . /etc/bash_completion
-fi
-
-# Some distribution makes use of a profile.d script to import completion.
-if [ -f /etc/profile.d/bash_completion.sh ]; then
-  . /etc/profile.d/bash_completion.sh
-fi
+# If bash-completion is already enabled (by e.g. /etc/bash.bashrc sourced from
+# /etc/profile or directly executed by Bash), we skip loading bash-completion.
+[[ ${BASH_COMPLETION_VERSINFO-} ]] && return 0
 
+# We skip loading bash-completion in the POSIX mode.  Older versions of
+# bash-completion do not work in the POSIX mode.
+shopt -oq posix && return 0
 
-if [ $(uname) = "Darwin" ] && _omb_util_command_exists brew; then
+# If Homebrew is installed (OS X), its Bash completion module is loaded.
+if is_os darwin && _omb_util_command_exists brew; then
   BREW_PREFIX=$(brew --prefix)
 
-  if [ -f "$BREW_PREFIX"/etc/bash_completion ]; then
-    . "$BREW_PREFIX"/etc/bash_completion
+  # homebrew/versions/bash-completion2 (required for projects.completion.bash)
+  # is installed to this path
+  if [[ -f "$BREW_PREFIX"/share/bash-completion/bash_completion ]]; then
```

Comment:
One request I would make here is the ability to set a `BASH_COMPLETION_DIR`, and check it as a potential location.

Reply:
I'm not sure if that is useful. If you know the location of bash_completion and needs to specify it explicitly, how is it different from `source '<path to bash_completion>/bash_completion'`? A plugin that allows users to write

```bash
BASH_COMPLETION_DIR='<path to bash_completion>/bash_completion'

plugins+=(system)
```

instead of

```bash
source '<path to bash_completion>/bash_completion'
```

doesn't seem useful. It actually complicates the configuration.

Also, the plugin name `system` implies loading `bash-completion` in a system package. Allowing the configuration for an arbitrary location of `bash-completion` doesn't match the plugin's name. If the configuration would be added, the plugin name should also be changed.

In addition, `BASH_COMPLETION_*` is a kind of variable namespace used by `bash-completion`, so we should avoid defining a variable of the form `BASH_COMPLETION_*` outside `bash-completion`. It might conflict in the future.

https://github.com/scop/bash-completion/blob/main/doc/api-and-naming.md#naming
https://github.com/scop/bash-completion/blob/main/doc/configuration.md